src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -89,6 +89,12 @@ public OpenIdConnectHandler(HttpClient backchannel)
                     message.PostLogoutRedirectUri = Options.PostLogoutRedirectUri;
                 }
 
+                var ticket = await HandleAuthenticateOnceAsync();
+                if (ticket != null)
+                {
+                    message.IdTokenHint = ticket.Principal?.FindFirst(OpenIdConnectParameterNames.IdToken)?.Value;
```

Comment:
> There are btw situations where you don't want to send the id_token on logout - I had a couple of cases where it was too big for the query string and IE bombed.

True. Maybe a new option - `FlowIdentityTokenHint` - would help mitigating this issue?

FYI: this middleware now supports POST logout requests via an auto-post form, which solves this kind of issues. But since it's not supported everywhere, I guess the additional option is worth it.